ts/main.ts

```diff
@@ -243,3 +250,52 @@ $entriesList.addEventListener('click', function (event: Event) {
     $notes.value = data.editing.notes;
   }
 });
+
+const $deleteButton = document.querySelector(
+  '.delete-button',
+) as HTMLButtonElement;
+if (!$deleteButton) throw new Error('.delete-button query failed!');
+const $modal = document.querySelector('dialog') as HTMLDialogElement;
+if (!$modal) throw new Error('dialog query failed!');
+const $dismissModal = document.querySelector(
+  '.dismiss-modal',
+) as HTMLButtonElement;
+if (!$dismissModal) throw new Error('.dismiss-modal query failed!');
+const $confirmDelete = document.querySelector(
+  '.confirm-delete',
+) as HTMLButtonElement;
+if (!$confirmDelete) throw new Error('.confirm-delete query failed!');
+
+$deleteButton.addEventListener('click', function () {
+  $modal.showModal();
+});
+
+$dismissModal.addEventListener('click', function (event: Event) {
+  event.preventDefault();
+  console.log('CANCEL BUTTON PRESSED');
+  $modal.close();
```

Comment:
Remove `console.log`

Also, you shouldn't need the `event.preventDefault()` here because you have already changed your button to `type='button'`. The reason we use it is to prevent the default method which for a form it prevents the default submission. This would maybe be needed on a button in a form because the default for a button is to submit, but already assigning `type='button'` would disable the submission anyways.